# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: cb0e58368cd4
# Full Hash: cb0e58368cd4d8e1f79b193b7ccae9586cc2ab50
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 10: Use TaggedParserAtomIndex in StencilModuleEntry. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96917
# ==============================================================================

diff -r 73cc5fa0e0bd -r cb0e58368cd4 js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:20 2020 +0000
+++ b/js/src/frontend/Stencil.cpp	Sat Nov 14 00:03:28 2020 +0000
@@ -1238,28 +1238,29 @@
 }
 
 static void DumpModuleEntryVectorItems(
-    js::JSONPrinter& json, const StencilModuleMetadata::EntryVector& entries) {
+    js::JSONPrinter& json, const StencilModuleMetadata::EntryVector& entries,
+    CompilationStencil* compilationStencil) {
   for (const auto& entry : entries) {
     json.beginObject();
     if (entry.specifier) {
-      GenericPrinter& out = json.beginStringProperty("specifier");
-      entry.specifier->dumpCharsNoQuote(out);
-      json.endStringProperty();
+      json.beginObjectProperty("specifier");
+      DumpTaggedParserAtomIndex(json, entry.specifier, compilationStencil);
+      json.endObject();
     }
     if (entry.localName) {
-      GenericPrinter& out = json.beginStringProperty("localName");
-      entry.localName->dumpCharsNoQuote(out);
-      json.endStringProperty();
+      json.beginObjectProperty("localName");
+      DumpTaggedParserAtomIndex(json, entry.localName, compilationStencil);
+      json.endObject();
     }
     if (entry.importName) {
-      GenericPrinter& out = json.beginStringProperty("importName");
-      entry.importName->dumpCharsNoQuote(out);
-      json.endStringProperty();
+      json.beginObjectProperty("importName");
+      DumpTaggedParserAtomIndex(json, entry.importName, compilationStencil);
+      json.endObject();
     }
     if (entry.exportName) {
-      GenericPrinter& out = json.beginStringProperty("exportName");
-      entry.exportName->dumpCharsNoQuote(out);
-      json.endStringProperty();
+      json.beginObjectProperty("exportName");
+      DumpTaggedParserAtomIndex(json, entry.exportName, compilationStencil);
+      json.endObject();
     }
     json.endObject();
   }
@@ -1268,34 +1269,36 @@
 void StencilModuleMetadata::dump() {
   js::Fprinter out(stderr);
   js::JSONPrinter json(out);
-  dump(json);
+  dump(json, nullptr);
 }
 
-void StencilModuleMetadata::dump(js::JSONPrinter& json) {
+void StencilModuleMetadata::dump(js::JSONPrinter& json,
+                                 CompilationStencil* compilationStencil) {
   json.beginObject();
-  dumpFields(json);
+  dumpFields(json, compilationStencil);
   json.endObject();
 }
 
-void StencilModuleMetadata::dumpFields(js::JSONPrinter& json) {
+void StencilModuleMetadata::dumpFields(js::JSONPrinter& json,
+                                       CompilationStencil* compilationStencil) {
   json.beginListProperty("requestedModules");
-  DumpModuleEntryVectorItems(json, requestedModules);
+  DumpModuleEntryVectorItems(json, requestedModules, compilationStencil);
   json.endList();
 
   json.beginListProperty("importEntries");
-  DumpModuleEntryVectorItems(json, importEntries);
+  DumpModuleEntryVectorItems(json, importEntries, compilationStencil);
   json.endList();
 
   json.beginListProperty("localExportEntries");
-  DumpModuleEntryVectorItems(json, localExportEntries);
+  DumpModuleEntryVectorItems(json, localExportEntries, compilationStencil);
   json.endList();
 
   json.beginListProperty("indirectExportEntries");
-  DumpModuleEntryVectorItems(json, indirectExportEntries);
+  DumpModuleEntryVectorItems(json, indirectExportEntries, compilationStencil);
   json.endList();
 
   json.beginListProperty("starExportEntries");
-  DumpModuleEntryVectorItems(json, starExportEntries);
+  DumpModuleEntryVectorItems(json, starExportEntries, compilationStencil);
   json.endList();
 
   json.beginListProperty("functionDecls");
@@ -1651,7 +1654,7 @@
 
   if (scriptData[CompilationInfo::TopLevelIndex].isModule()) {
     json.beginObjectProperty("moduleMetadata");
-    moduleMetadata.dumpFields(json);
+    moduleMetadata.dumpFields(json, this);
     json.endObject();
   }
 